pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -203,6 +203,34 @@ func (builder *ListenerBuilder) buildVirtualInboundListener(env *model.Environme
 	return builder
 }
 
+func (builder *ListenerBuilder) patchListeners(push *model.PushContext) {
+	if builder.node.Type == model.Router {
+		envoyfilter.ApplyListenerPatches(networking.EnvoyFilter_GATEWAY, builder.node, push, builder.gatewayListeners, false)
+		return
+	}
+
+	patchOneListener := func(listener *xdsapi.Listener) *xdsapi.Listener {
+		if listener == nil {
+			return nil
+		}
+		tempArray := []*xdsapi.Listener{listener}
+		tempArray = envoyfilter.ApplyListenerPatches(networking.EnvoyFilter_SIDECAR_OUTBOUND, builder.node, push, tempArray, true)
```

Comment:
This lambda is used by both in and out bound virtual listener. Why it is the `Apply` always takes `EnvoyFilter_SIDECAR_OUTBOUND`?